### PR TITLE
Updated Resources SDK to version 2.0.0-preview

### PR DESCRIPTION
**Pull request by luanshixia**
Updated Resources SDK to version 2.0.0-preview.

## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed. You can find a more complete discussion of PowerShell cmdlet best practices [here](https://msdn.microsoft.com/en-us/library/dd878270(v=vs.85).aspx).

- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md).**
- [x] **If changes were made to any cmdlet, the XML help was regenerated using the [platyPSHelp module](https://github.com/Azure/azure-powershell/blob/dev/documentation/platyPSHelp-documentation.md).**
- [x] **If any large changes are made to a service, they are reflected in the respective [change log](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#updating-the-change-log).**

### [General Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [x] The pull request does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/dev/documentation/breaking-changes.md) (unless a major version change occurs in the assembly and module).

### [Testing Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.
- [x] PowerShell scripts used in tests should do any necessary setup as part of the test or suite setup, and should not use hard-coded values for locations or existing resources.

### [Cmdlet Signature Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#cmdlet-signature-guidelines)
- [x] New cmdlets that make changes or have side effects should implement `ShouldProcess` and have `SupportShouldProcess=true` specified in the cmdlet attribute. You can find more information on `ShouldProcess` [here](https://gist.github.com/markcowl/338e16fe5c8bbf195aff9f8af0db585d#what-is-the-change).
- [x] Cmdlet specifies `OutputType` attribute if any output is produced - if the cmdlet produces no output, it should implement a `PassThru` parameter.

### [Cmdlet Parameter Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#cmdlet-parameter-guidelines)
- [x] Parameter types should not expose types from the management library - complex parameter types should be defined in the module.
- [x] Complex parameter types are discouraged - a parameter type should be simple types as often as possible. If complex types are used, they should be shallow and easily creatable from a constructor or another cmdlet.
- [x] Cmdlet parameter sets should be mutually exclusive - each parameter set must have at least one mandatory parameter not in other parameter sets.

### CODE REVIEW

File: src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Commands.Resources.Rest.csproj [hunk 1]
```diff
@@ -54,9 +54,8 @@
       <SpecificVersion>False</SpecificVersion>
       <HintPath>..\..\..\..\packages\Microsoft.Azure.Common.2.1.0\lib\net45\Microsoft.Azure.Common.dll</HintPath>
     </Reference>
-    <Reference Include="Microsoft.Azure.Management.ResourceManager, Version=1.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
-      <SpecificVersion>False</SpecificVersion>
-      <HintPath>..\..\..\..\packages\Microsoft.Azure.Management.ResourceManager.1.2.0-preview\lib\net45\Microsoft.Azure.Management.ResourceManager.dll</HintPath>
+    <Reference Include="Microsoft.Azure.Management.ResourceManager">
+      <HintPath>..\..\..\..\packages\Microsoft.Azure.Management.ResourceManager.2.0.0-preview\lib\net452\Microsoft.Azure.Management.ResourceManager.dll</HintPath>
     </Reference>
```
Comment (luanshixia) [inline src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Commands.Resources.Rest.csproj]: Right now this is from local built library. Will change to strong named assembly after SDK merged and new NuGet version dropped.

**Comment by luanshixia** [comment]
Test recordings were not updated because the service is still in deployment progress. I will update them later.

**Comment by markcowl** [comment]
@luanshixia Is this feature guaranteed to make it in for //build?  Can you record the tests against a test cluster?

**Comment by markcowl** [comment]
@luanshixia This is failing because the package is not published.  If this feature is not guaranteed to be available for BUILD, please close this PR, and open a new one after the BUILD release is out.

**Comment by luanshixia** [comment]
@markcowl This feature has to be in for //build, that's why we were in such a hurry... I will commit the recorded tests today.

File: src/ResourceManager/Resources/Resources.sln [hunk 2]
```diff
@@ -1,6 +1,6 @@
 Microsoft Visual Studio Solution File, Format Version 12.00
 # Visual Studio 14
-VisualStudioVersion = 14.0.25123.0
+VisualStudioVersion = 14.0.25420.1
```
Comment (TianoMS) [inline src/ResourceManager/Resources/Resources.sln]: >14.0.25420.1 [](start = 22, length = 12)

do we need this change?

Comment (luanshixia) [inline src/ResourceManager/Resources/Resources.sln]: I can revert this.

File: src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Implementation/Providers/GetAzureProviderCmdlet.cs [hunk 3]
```diff
@@ -136,16 +136,15 @@ private PSResourceProvider[] ListPSResourceProviders()
                 return new PSResourceProvider[] { };
             }
 
-            foreach (var provider in providers)
-            {
-                provider.ResourceTypes = provider.ResourceTypes
+            var providerWithResourceTypes = providers.ToDictionary(
+                provider => provider, 
+                provider => provider.ResourceTypes
```
Comment (TianoMS) [inline src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Implementation/Providers/GetAzureProviderCmdlet.cs]: do we need this change?

File: src/ResourceManager/Resources/Commands.Resources.Test/Commands.Resources.Test.csproj [hunk 4]
```diff
@@ -369,31 +379,31 @@
       <CopyToOutputDirectory>Always</CopyToOutputDirectory>
     </None>
     <None Include="SessionRecords\Microsoft.Azure.Commands.Resources.Test.ScenarioTests.DeploymentTests\TestNestedDeploymentFromTemplateFile.json">
-      <CopyToOutputDirectory>Always</CopyToOutputDirectory>
+      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
```
Comment (TianoMS) [inline src/ResourceManager/Resources/Commands.Resources.Test/Commands.Resources.Test.csproj]: ><CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory> [](start = 6, length = 61)

revert these changes?

File: src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Implementation/Providers/GetAzureProviderCmdlet.cs (hunk 3 above)
Comment (luanshixia) [inline src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Implementation/Providers/GetAzureProviderCmdlet.cs]: @TianoMS This was due to `ResourceTypes` property is made `readonly` now.

File: src/ResourceManager/Resources/Commands.Resources.Test/Models.ResourceGroups/ResourceClientTests.cs [hunk 5]
```diff
@@ -51,7 +51,7 @@ public class ResourceClientTests : RMTestBase
 
         private Mock<IResourcesOperations> resourceOperationsMock;
 
-        private Mock<IDeploymentOperationsOperations> deploymentOperationsMock;
+        private Mock<IDeploymentOperations> deploymentOperationsMock;
```
Comment (TianoMS) [inline src/ResourceManager/Resources/Commands.Resources.Test/Models.ResourceGroups/ResourceClientTests.cs]: >IDeploymentOperations [](start = 21, length = 21)

Is this the right change? Please check SDK classes.

**Comment by TianoMS** [comment]
> {

Do we need this file?

---
Refers to: src/ResourceManager/Resources/Commands.Resources.Test/SessionRecords/Microsoft.Azure.Commands.Resources.Test.ScenarioTests.DeploymentTests/TestCrossResourceGroupDeploymentFromTemplateFile1.json:1 in b4654c9. [](commit_id = b4654c9ba35ac0ed941d44792ae462bb61624ee3, deletion_comment = False)

File: src/ResourceManager/Resources/Commands.Resources.Test/Models.ResourceGroups/ResourceClientTests.cs (hunk 5 above)
Comment (luanshixia) [inline src/ResourceManager/Resources/Commands.Resources.Test/Models.ResourceGroups/ResourceClientTests.cs]: @TianoMS I believe this was due to code generation was revised.

**Comment by luanshixia** [comment]
@markcowl I will update the changelog. No output types have changed.

**Comment by markcowl** [comment]
on demand run here:  http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/1546/

File: src/ResourceManager/Resources/Commands.Resources.Test/ScenarioTests/DeploymentTests.cs [hunk 6]
```diff
@@ -44,6 +44,12 @@ public void TestNestedDeploymentFromTemplateFile()
         }
 
         [Fact]
+        public void TestCrossResourceGroupDeploymentFromTemplateFile()
```
Comment (cormacpayne) [inline src/ResourceManager/Resources/Commands.Resources.Test/ScenarioTests/DeploymentTests.cs]: @luanshixia this test failed the on-demand job. Please make this test a check-in test so it is run on the regular job.

**Comment by cormacpayne** [comment]
On-demand: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/1555/

LGTM once on-demand passes
